src/contrib/cluster/Akka.DistributedData.Tests.MultiNode/ReplicatorSpec.cs

```diff
@@ -401,22 +421,24 @@ public void Cluster_CRDT_should_converge_after_partition()
             RunOn(() =>
             {
                 _replicator.Tell(Dsl.Get(KeyD, ReadLocal.Instance));
-                var c40 = ExpectMsg<Replicator.GetSuccess>(g => Equals(g.Key, KeyD)).Get(KeyD);
+                var c40 = ExpectMsg<GetSuccess>(g => Equals(g.Key, KeyD)).Get(KeyD);
                 c40.Value.ShouldBe(40);
 
-                _replicator.Tell(Dsl.Update(KeyD, GCounter.Empty.Increment(_cluster.SelfUniqueAddress, 1), _writeTwo, x => x.Increment(_cluster.SelfUniqueAddress, 1)));
-                ExpectMsg(new Replicator.UpdateTimeout(KeyD, null), _timeOut.Add(TimeSpan.FromSeconds(1)));
-                _replicator.Tell(Dsl.Update(KeyD, GCounter.Empty, _writeTwo, x => x.Increment(_cluster.SelfUniqueAddress, 1)));
-                ExpectMsg(new Replicator.UpdateTimeout(KeyD, null), _timeOut.Add(TimeSpan.FromSeconds(1)));
+                _replicator.Tell(Dsl.Update(KeyD, GCounter.Empty.Increment(_cluster, 1), _writeTwo, x => x.Increment(_cluster, 1)));
+                ExpectMsg(new UpdateTimeout(KeyD, null), _timeOut.Add(TimeSpan.FromSeconds(1)));
+                _replicator.Tell(Dsl.Update(KeyD, GCounter.Empty, _writeTwo, x => x.Increment(_cluster, 1)));
+                ExpectMsg(new UpdateTimeout(KeyD, null), _timeOut.Add(TimeSpan.FromSeconds(1)));
             }, _config.First, _config.Second);
 
             RunOn(() =>
             {
+                //TODO: for some reason this RunOn never gets called
                 for (int i = 1; i <= 30; i++)
                 {
-                    var keydn = new GCounterKey("D" + i);
-                    _replicator.Tell(Dsl.Update(keydn, GCounter.Empty, WriteLocal.Instance, x => x.Increment(_cluster.SelfUniqueAddress, i)));
-                    ExpectMsg(new Replicator.UpdateSuccess(keydn, null));
+                    var n = i;
+                    var keydn = new GCounterKey("D" + n);
+                    _replicator.Tell(Dsl.Update(keydn, GCounter.Empty, WriteLocal.Instance, x => x.Increment(_cluster, n)));
+                    ExpectMsg(new UpdateSuccess(keydn, null));
                 }
             }, _config.First);
```

Comment:
I'm not sure if this is 100% reproducible in each case but it looks like this pattern is a bug in MNTK:

```csharp
EnterBarrier("after-1");

RunOn(() => {
    // this one gets called
}, firstRole, secondRole);

RunOn(() => {
    // this one never gets called
}, firstRole);

EnterBarrier("after-2");
```